src/App.js

```diff
@@ -25,7 +29,24 @@ class App extends Component {
                 { name: 'Jared', id: 0 },
                 { name: 'Adrian', id: 1 }
             ],
+            addingPlayer: false
         }
+        this.id = 1
+        this.addPlayer = this.addPlayer.bind(this);
+        this.addingPlayer = this.addingPlayer.bind(this);
```

Comment:
using bind makes for a lot of unneeded boilerplate in the constructor its practice now a days to just use arrow syntax on your methods 
```
addingPlayer = (addingPlayer) => {...}
```
Fat arrow functions automatically bind the scope of their parent so there's no need for bind.